### PR TITLE
[openimageio] Update to v2.4.5.0

### LINKED ISSUE #27666: [openimageio] update to <v2.4.5.0>
Library name: OpenImageIO

New version number: v2.4.5.0

Other information that may be useful (release notes, etc...)
https://github.com/OpenImageIO/oiio/releases/tag/v2.4.5.0

### PR DESCRIPTION
**Pull request by MonicaLiu0311**
Fixes https://github.com/microsoft/vcpkg/issues/27666. 
Update version to v2.4.5.0.

All features are tested successfully in the following triplet:
x86-windows
x64-windows
x64-windows-static

### CODE REVIEW

**Review by dg0yt — CHANGES_REQUESTED** [review]
Unless there are good reasons, can you please restore the original patch structure, for the benefit of PR review and `git blame`?

File: ports/openimageio/fix-dependencies.patch [hunk 1]
```diff
@@ -1,106 +1,95 @@
...
+-checked_find_package (OpenColorIO
++set(OPENCOLORIO_INCLUDES "")
++set(OPENCOLORIO_LIBRARIES OpenColorIO::OpenColorIO)
++checked_find_package (OpenColorIO CONFIG 
+                       DEFINITIONS  -DUSE_OCIO=1 -DUSE_OPENCOLORIO=1
+                       # PREFER_CONFIG
+                       )
++set(OPENCOLORIO_FOUND "${OpenColorIO_FOUND}")
+ if (NOT OpenColorIO_FOUND)
+     set (OpenColorIO_FOUND 0)
```
Comment (dg0yt) [inline ports/openimageio/fix-dependencies.patch:45]: Is `OPENCOLORIO_FOUND` spelling still used? If not, we don't need to inject it.

File: ports/openimageio/fix-dependencies.patch [hunk 2]
```diff
@@ -1,106 +1,95 @@
...
+ 
+ # Intel TBB
+@@ -191,14 +196,18 @@ checked_find_package (TBB 2017
+                       PREFER_CONFIG)
+ 
+ checked_find_package (DCMTK VERSION_MIN 3.6.1)  # For DICOM images
+-checked_find_package (FFmpeg VERSION_MIN 3.0)
++checked_find_package (FFmpeg)
++set(FFmpeg_FOUND "${FFMPEG_FOUND}")
++set(FFMPEG_INCLUDES "${FFMPEG_INCLUDE_DIRS}")
```
Comment (dg0yt) [inline ports/openimageio/fix-dependencies.patch:61]: Which `FFmpeg` package is loaded now? Does it set `FFmpeg_FOUND` or `FFMPEG_FOUND`?

File: ports/openimageio/portfile.cmake [hunk 3]
```diff
@@ -59,13 +59,14 @@ vcpkg_cmake_configure(
         -DVERBOSE=ON
         -DBUILD_DOCS=OFF
         -DINSTALL_DOCS=OFF
+		-DENABLE_INSTALL_testtex=OFF
```
Comment (dg0yt) [inline ports/openimageio/portfile.cmake:62]: Indent...

File: versions/o-/openimageio.json [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "d5fa430486d400bbbb2411f8e21b98c460b39c00",
+      "version": "2.4.5.0",
+      "port-version": 0
+    },
+    {
+      "git-tree": "86a3e95dda74f013b6c55a0ef923769425cbe8fb",
+      "version": "2.4.5.0",
+      "port-version": 4
+    },
```
Comment (dg0yt) [inline versions/o-/openimageio.json:12]: No.

**Review by dg0yt — CHANGES_REQUESTED** [review]
`fix-dependencies.patch`seems to have the line breaks changed.

File: ports/openimageio/fix-dependencies.patch [hunk 5]
```diff
@@ -21,14 +21,15 @@ index b690864..fb45388 100644
 +if(@USE_WEBP@)
 +    find_dependency(WebP CONFIG)
 +endif()
- 
++
  # add here all the find_dependency() whenever switching to config based dependencies
- if (@OpenEXR_VERSION@ VERSION_GREATER_EQUAL 3.0)
+ if (NOT @OPENIMAGEIO_CONFIG_DO_NOT_FIND_IMATH@ AND NOT OPENIMAGEIO_CONFIG_DO_NOT_FIND_IMATH)
+     if (@OpenEXR_VERSION@ VERSION_GREATER_EQUAL 3.0)
```
Comment (JonLiu1993) [inline ports/openimageio/fix-dependencies.patch:27]: fjggj

File: ports/openimageio/portfile.cmake [hunk 6]
```diff
@@ -1,14 +1,15 @@
...
-    SHA512 25cb1a671e7cd5154e363eef178ab091fd7d55868746a4394340567a794f6c6f0295e58721a5b4ee8bf66b4cc0e6a01c3e82f9cc9de9953ae349d45738a04700
+    REF 1e70f7862ceca56c0d30185837ea08e765fa73db    # v2.4.5.0
+    SHA512 f0336b1f5de4709cf7d7194e55b9f54d76261287e0ddc38b9a8605a0a3383f271b5cb605026b3911c48e90087a4574fe62dfa911edbe1cb3570dff380178154b
     HEAD_REF master
     PATCHES
         fix-dependencies.patch
         fix-static-ffmpeg.patch
         fix-openexr-dll.patch
         imath-version-guard.patch
+		fix-openimageio_include_dir.patch
```
Comment (JonLiu1993) [inline ports/openimageio/portfile.cmake:12]:
```suggestion
        imath-version-guard.patch
	fix-openimageio_include_dir.patch
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — DISMISSED** [review]

File: ports/openimageio/portfile.cmake [hunk 7]
```diff
@@ -1,14 +1,15 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO OpenImageIO/oiio
-    REF v2.3.17.0
-    SHA512 25cb1a671e7cd5154e363eef178ab091fd7d55868746a4394340567a794f6c6f0295e58721a5b4ee8bf66b4cc0e6a01c3e82f9cc9de9953ae349d45738a04700
+    REF 1e70f7862ceca56c0d30185837ea08e765fa73db    # v2.4.5.0
```
Comment (dg0yt) [inline ports/openimageio/portfile.cmake:4]: Don't you want to use `v${VERSION}`?
There was an explict version tag reference before, and this PR replaces it with an commit ID. I already considered this bad practice in the past, and I just fixed another popular port where the ref didn't match the claimed version.

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/openimageio/portfile.cmake [hunk 8]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
```
Comment (JonLiu1993) [inline ports/openimageio/portfile.cmake:1]:
```suggestion
vcpkg_minimum_required(VERSION 2022-10-12)
vcpkg_from_github(
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**